pageserver/src/walredo.rs

```diff
@@ -538,165 +521,142 @@ impl PostgresRedoProcess {
             datadir.display()
         );
 
-        let stdin = child.stdin.take().expect("failed to open child's stdin");
-        let stderr = child.stderr.take().expect("failed to open child's stderr");
-        let stdout = child.stdout.take().expect("failed to open child's stdout");
+        let stdin = child.stdin.take().unwrap();
+        let stdout = child.stdout.take().unwrap();
+        let stderr = child.stderr.take().unwrap();
 
-        // This async block reads the child's stderr, and forwards it to the logger
-        let f_stderr = async {
-            let mut stderr_buffered = tokio::io::BufReader::new(stderr);
+        nix::fcntl::fcntl(stdin.as_raw_fd(), F_SETFL(OFlag::O_NONBLOCK))?;
+        nix::fcntl::fcntl(stdout.as_raw_fd(), F_SETFL(OFlag::O_NONBLOCK))?;
+        nix::fcntl::fcntl(stderr.as_raw_fd(), F_SETFL(OFlag::O_NONBLOCK))?;
```

Comment:
Are there any flags that will be unset by doing this? The typical pattern I see is calling F_GETFL and then setting the NONBLOCK bit

Reply:
Good point. Changed it that way. I added a little utility function to do the F_GETFL + F_SETFL dance in zenith_utils, it might come handy elsewhere too.